bndtools.core/src/bndtools/central/Central.java

```diff
@@ -204,39 +203,52 @@ public static Workspace getWorkspace() throws Exception {
             if (ws != null) { // early check for workspace
                 return ws;
             }
-            try {
-                Workspace.setDriver(Constants.BNDDRIVER_ECLIPSE);
-                Workspace.addGestalt(Constants.GESTALT_INTERACTIVE, new Attrs());
+            ws = initializeWorkspace();
+            // The workspace has been initialized fully, set the field now
+            workspace = ws;
+        }
+        workspaceQueue.resolve(ws); // notify onWorkspaceInit callbacks
```

Comment:
> As far as I understand the Promise, workspaceQueue is resolved with the new workspace instance, not the old one.

workspaceQueue is a final variable and a Promise can be resolved exactly once. So once the promise is resolved with a workspace, all callbacks will use that workspace.


> EDIT: I've added an additional workspaceQueue.resolve in resetWorkspace

A second call to resolve will throw an IllegalArgumentException.